File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.py
def solution(a, b):
    answer = 0
    min = 0
    max = 0
    if a == b :
        return a
    elif a > b :
        max = a
        min = b
    else:
        max = b
        min = a
        
    for i in range(min, max+1):
        answer += i
        
    
    return answer

File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int a, int b) {
    long long answer = 0;
    int x, y;
    if (a > b) {
        x = b;
        y = a;
    }
    else {
        x = a;
        y = b;
    }
    
    long long k = y-x+1; 
    if (k%2 == 1) {
        answer = k/2 * (y + x) + (y+x)/2;
    }
    else {
        answer = k/2 * (y + x);
    }
    return answer;
}